src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateSet.YYYY) {
+    dateSet.YY = dateSet.YYYY.slice(2);
+  }
+
+  if (dateSet.YY < 30) {
```

Comment:
Direct comparison of 'dateSet.YY' as a string with a number may lead to unexpected results due to type coercion. Ensure that 'dateSet.YY' is converted to a number before comparing it with 30.